### ISSUE TITLE
Support for Text Objects

### ISSUE THREAD

**Issue by techniq** [aioutecism/amVim-for-VSCode#13]
I'm creating an issue to track this feature.  See **Text objects (only in Visual mode or after an operator)** in issue #1

These are a few of my commonly used ones:
- `ciw` - change in word
- `viw` - visual select in word
  - I typically use this followed with yank and paste
- `ci(` or `ci)` - change in parens
  - Useful to change function arguments or if/switch conditions, etc
- `ci<` or `ci>` - change in angle brackets
  - Useful to change html, for example, change `<div>` to `<span>`
- `cit` - change in tag
  - Example `<p>Some | text</p>`: If cursor is `l`, `ciw` will replace `Some text` inside `<p></p>`

**Comment by aioutecism** [comment]
Support text objects needs some refactoring.
I'll get to this one later.

**Comment by Raynes** [comment]
Desperately want this. I'll name my first born after you if you do it.

**Comment by Beatusvir** [comment]
Don't foget:
* `ct`X - change till X, moves forward
* `cT`X - change till X, moves backward
* `cf`X - change till X (including X), moves forward
* `cF`X - change till X (including X), moves backward

**Comment by aioutecism** [comment]
Parts of text objects are supported in 1.3.1.
See #1 - "Text objects (only in Visual mode or after an operator)" for detail.

**Comment by aioutecism** [comment]
@Beatusvir "Till" is not a "text object" and is supported several versions before.

**Comment by gentius** [comment]
Just want to say thanks for the partial support for text objects, the stuff that is in there seems to work well.

I'll add another vote for the visual select stuff which is still missing like `viB`, `vi(`, etc.
Useful for confirming the operation in some situations instead of going straight ahead with `c` or `d`.

**Comment by aioutecism** [comment]
@gentius 
Yes, visual mode support and word/sentence/paragraph support is missing now.
Going to add them soon.

**Comment by skube** [comment]
I like to use `vip` and `vap` which prevents me from using this particular extension.

**Comment by alisonatwork** [comment]
Version 1.31.0 has just been released today which enables tag-style text objects (`cit`, `dit` etc). Please give it a test run and let us know how it works for you.
